config/yaml_config.go

```diff
@@ -75,19 +77,33 @@ func (a *AgentConfig) loadNetworkYamlConfig(path string) error {
 	// Pull additional parameters from the global config file.
 	a.LogLevel = config.Datadog.GetString("log_level")
 	a.StatsdPort = config.Datadog.GetInt("dogstatsd_port")
+	if config.Datadog.IsSet("bind_host") {
+		a.StatsdHost = config.Datadog.GetString("bind_host")
+	}
 
 	return nil
 }
 
 // Process-specific configuration
-func (a *AgentConfig) loadProcessYamlConfig(path string) error {
-	config.Datadog.AddConfigPath(path)
-	if strings.HasSuffix(path, ".yaml") { // If they set a config file directly, let's try to honor that
-		config.Datadog.SetConfigFile(path)
-	}
+func (a *AgentConfig) loadProcessConfig(iniPath, yamlPath string) error {
```

Comment:
Should we add a check to make sure iniPath and yamlPath are not both set, since this will behave unexpectedly in case they are. Or even accept one string and infer the type based on the extension?

Reply:
The behavior will change to prefer the YAML if both exist, similar to how the main + trace agents operate. We're going to list this as a breaking change in our release notes.